java/client/src/main/java/glide/managers/CommandManager.java

```diff
@@ -31,8 +36,35 @@ public <T> CompletableFuture<T> submitNewCommand(
     // create protobuf message from command
     // submit async call
     return channel
-        .write(prepareRedisRequest(command.getRequestType(), command.getArguments()), true)
-        .thenApplyAsync(response -> responseHandler.apply(response));
+        .write(
+            RedisRequest.newBuilder()
+                .setSingleCommand(
+                    prepareRedisCommand(command.getRequestType(), command.getArguments())),
+            true)
+        .thenApplyAsync(responseHandler::apply);
+  }
+
+  /**
+   * Build a transaction and send.
+   *
+   * @param transaction The command to execute
+   * @param responseHandler The handler of the response object
+   * @return A result promise of type T
+   */
+  public <T> CompletableFuture<T> submitNewTransaction(
+      List<Command> transaction, RedisExceptionCheckedFunction<Response, T> responseHandler) {
+    // register callback
+    // create protobuf message from command
+    // submit async call
+    var transactionBuilder = Transaction.newBuilder();
+    for (var command : transaction) {
+      transactionBuilder.addCommands(
+          prepareRedisCommand(command.getRequestType(), command.getArguments()));
```

Comment:
how does this work routes?  Each command gets a route, right?  
I think the API.Command object can be used to pass that in, and there will be a third argument in prepareRedisCommand that takes a `command.getRouteOption()`, right?

Reply:
According to protobuf - route set to the entire request